lib/SILOptimizer/Mandatory/DIMemoryUseCollector.cpp

```diff
@@ -1244,13 +1241,27 @@ ElementUseCollector::collectAssignOrInitUses(AssignOrInitInst *Inst,
   /// that the flag is dropped before calling \c addElementUses.
   llvm::SaveAndRestore<bool> X(IsSelfOfNonDelegatingInitializer, false);
 
-  auto *typeDC = Inst->getReferencedInitAccessor()
-                     ->getDeclContext()
-                     ->getSelfNominalTypeDecl();
+  NominalTypeDecl *typeDC = nullptr;
+  if (auto accessorDecl = Inst->getReferencedInitAccessor()) {
+    typeDC = accessorDecl->getDeclContext()->getSelfNominalTypeDecl();
+  } else {
+    typeDC = Inst->getProperty()->getDeclContext()->getSelfNominalTypeDecl();
```

Comment:
I think this could be wrapped in a method on AssignOrInit instruction which is going to produce a decl context or a nullptr and you can just use `if (auto *typeDC = Inst->getXXX())` below.